monix-reactive/shared/src/main/scala/monix/reactive/Observable.scala

```diff
@@ -682,6 +682,23 @@ object Observable {
   def cons[A](head: A, tail: Observable[A]): Observable[A] =
     new builders.ConsObservable[A](head, tail)
 
+  /** Creates a new observable from this observable and another given
+    * observable by interleaving their items into a strictly alternating sequence.
+    *
+    * So the first item emitted by the new observable will be the item whichever
+    * child stream emits first, and the second item will be emitted by the other
```

Comment:
Wait - I thought the result is *deterministic*. This description says that whomever emits first, that's going to be the first stream. I don't think that's desirable in this case, I think the order should be strict, with the result deterministic (first stream emits first, second emits second).

The rationale is this: think of `zip` versus `combineLatest`. The `combineLatest` operator is like a `zip` that emits items whenever one of the streams emits and it doesn't pause any of them so you get a merge that's going to emit at the same pace as the fastest of the streams or even faster. Whereas with `zip`, in order to keep the result deterministic and get the first item pair joined, the second item pair, etc. we end up pausing the stream that's fastest, we make *a sacrifice*.

OK, so in other words - with non-deterministic operators, you need to gain something, like speed (e.g. `merge` versus `concat` / `flatMap`). But `interleave`  behaves on one hand very much like `zip`. So we are *rushing* to emit the first element, but for what? After the first element we are going to end up blocking and pausing so there's no gain. And that initial race condition buys us nothing IMHO.